src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -0,0 +1,131 @@
+                }
+            }
+            else
+            {
+                ReadOnlySpan<byte> propertyName = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+                if (reader._stringHasEscaping)
+                {
+                    int idx = propertyName.IndexOf(JsonConstants.BackSlash);
+                    Debug.Assert(idx != -1);
+                    propertyName = GetUnescapedString(propertyName, idx);
+                }
+
+                state.Current.JsonPropertyInfo = state.Current.JsonClassInfo.GetProperty(options, propertyName, ref state.Current);
+                if (state.Current.JsonPropertyInfo == null)
+                {
+                    if (state.Current.JsonClassInfo.DataExtensionProperty == null)
+                    {
+                        state.Current.JsonPropertyInfo = JsonPropertyInfo.s_missingProperty;
+                    }
+                    else
+                    {
+                        ProcessMissingProperty(propertyName, options, ref reader, ref state);
+                    }
+                }
+                else
+                {
+                    state.Current.PropertyIndex++;
+                }
+            }
+        }
+
+        private static void ProcessMissingProperty(
+            ReadOnlySpan<byte> unescapedPropertyName,
+            JsonSerializerOptions options,
+            ref Utf8JsonReader reader,
+            ref ReadStack state)
+        {
+            JsonPropertyInfo jsonPropertyInfo = state.Current.JsonClassInfo.DataExtensionProperty;
+
+            Debug.Assert(jsonPropertyInfo != null);
+            Debug.Assert(state.Current.ReturnValue != null);
+
+            IDictionary extensionData = (IDictionary)jsonPropertyInfo.GetValueAsObject(state.Current.ReturnValue);
+            if (extensionData == null)
+            {
+                Type type = jsonPropertyInfo.DeclaredPropertyType;
+
+                // Create the appropriate dictionary type. We already verified the types.
+                Debug.Assert(type.IsGenericType);
+                Debug.Assert(type.GetGenericArguments().Length == 2);
+                Debug.Assert(type.GetGenericArguments()[0].UnderlyingSystemType == typeof(string));
+                Debug.Assert(
+                    type.GetGenericArguments()[1].UnderlyingSystemType == typeof(object) ||
+                    type.GetGenericArguments()[1].UnderlyingSystemType == typeof(JsonElement));
+
+                extensionData = (IDictionary)options.GetOrAddClass(type).CreateObject();
+                jsonPropertyInfo.SetValueAsObject(state.Current.ReturnValue, extensionData);
+            }
+
+            JsonElement jsonElement;
+            using (JsonDocument jsonDocument = JsonDocument.ParseValue(ref reader))
+            {
+                jsonElement = jsonDocument.RootElement.Clone();
+            }
+
+            string keyName = Encoding.UTF8.GetString(unescapedPropertyName);
```

Comment:
This overload doesn't exist on netstandard2.0. Hence the build failure on netfx:

> System\Text\Json\Serialization\JsonSerializer.Read.HandlePropertyName.cs(124,54): error CS1503: Argument 1: cannot convert from 'System.ReadOnlySpan<byte>' to 'byte[]'
System\Text\Json\Serialization\JsonSerializer.Write.HandleDictionary.cs(164,41): error CS1503: Argument 1: cannot convert from 'string' to 'System.ReadOnlySpan<char>'

Reply:
Yep. Thanks